implement #234 one way join code

Lucamuh opened the pull request.

Lucamuh committed:
Implement feature

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in bef6e9c according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/320

deepsource-autofix[bot] committed:
refactor: autofix issues in 1 file

Resolved issues in src/app/session/login/login.component.ts with DeepSource Autofix

Lucamuh committed:
make deepsource happy

Lucamuh committed:
Merge branch 'feature/234' of https://github.com/zskarte/zskarte-client into feature/234

gobeli commented on src/app/session/login/login.component.html:
```diff
@@ -32,17 +32,30 @@
+        </button>
+        <button class="alternative-login-button" *ngIf="isLoginWithCodeEnabled" mat-raised-button (click)="isLoginWithCodeEnabled = false">
+          {{ i18n.get('cancel') }}
+        </button>
```
Könnte man evtl. etwas vereinfachen.

```suggestion
        <button class="alternative-login-button" *ngIf="!isLoginWithCodeEnabled" mat-raised-button (click)="isLoginWithCodeEnabled = !isLoginWithCodeEnabled">
          <span *ngIf="!isLoginWithCodeEnabled">{{ i18n.get('codeLogin') }}</span>
          <span *ngIf="isLoginWithCodeEnabled">{{ i18n.get('cancel') }}</span>
        </button>
```

gobeli commented on src/app/session/login/login.component.ts:
```diff
@@ -87,7 +91,10 @@ export class LoginComponent {
+    if (this.isLoginWithCodeEnabled) {
+      const joinLink = `share/${this.joinCode}`;
+      await this.router.navigateByUrl(joinLink);
+    } else await this.session.login({ identifier: this.selectedOrganization?.identifier || '', password: this.password });
```
Ich würde hier für die readablility noch `{` um den else block machen.

Lucamuh replied:
```suggestion
    } else
    {
          await this.session.login({ identifier: this.selectedOrganization?.identifier || '', password: this.password });
    }
```

Lucamuh committed:
Improve readability

Co-authored-by: Etienne <etienne.gobeli@outlook.com>

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in d2d2053 according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/320

Lucamuh committed:
Improve readability

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 46fecd6 according to the output
from Prettier.

Details: https://github.com/zskarte/zskarte-client/pull/320

Lucamuh committed:
make deepsource happy

Lucamuh committed:
Merge branch 'feature/234' of https://github.com/zskarte/zskarte-client into feature/234
